### PR TITLE
Django 1.7 compatibility

### PR DESCRIPTION
**Pull request by gavinwahl**
This models breaks migrations in Django 1.7. We don't need to override this model in new versions of Django anyway. 

We can remove this file completely when we drop Django 1.4, but that's probably 9 months away if we follow Django's support policy.

### CODE REVIEW

**Comment by rockymeza** [comment]
I thought support for 1.4 would be dropped when 1.8 is released, seeing as it's the new LTS. Is this not true?

**Comment by gavinwahl** [comment]
The old LTS is supported for 6 months after the release of the new one.

**Comment by gavinwahl** [comment]
I'm updating this pull request to include all the django 1.7 commits. I've got it all working except for one thing I can't figure out.

Before, Content._meta.app_label was 'models', not 'widgy' like you might expect (it's abstract so it didn't really matter, except for template hierarchy). In django 1.7, with the new app loading, it's 'widgy'. TestTemplateHierarchy demonstrates the problem. Explicitly setting the app_label to widgy to use the new django behavior causes other tests to fail, so I'm not sure what to do.

**Comment by gavinwahl** [comment]
This is one way to fix it, but it seems pretty dirty:

```patch 
diff --git a/tests/modeltests/core_tests/tests/templates.py b/tests/modeltests/core_tests/tests/templates.py
index 4ee73cd..22021a8 100644
--- a/tests/modeltests/core_tests/tests/templates.py
+++ b/tests/modeltests/core_tests/tests/templates.py
@@ -114,10 +114,10 @@ class TestTemplateHierarchy(TestCase):
         self.assertEqual(list(unique_everseen(MyInvisibleBucket.get_templates_hierarchy(template_name='test'))), [
             'widgy/core_tests/myinvisiblebucket/test.html',
             'widgy/mixins/invisible/test.html',
-            'widgy/models/content/test.html',
+            'widgy/widgy/content/test.html',
             'widgy/core_tests/test.html',
             'widgy/mixins/test.html',
-            'widgy/models/test.html',
+            'widgy/widgy/test.html',
             'widgy/test.html',
         ])
 
@@ -126,8 +126,8 @@ class TestTemplateHierarchy(TestCase):
             'widgy/core_tests/weirdpkbucket/test.html',
             'widgy/core_tests/weirdpkbucketbase/test.html',
             'widgy/core_tests/weirdpkbase/test.html',
-            'widgy/models/content/test.html',
+            'widgy/widgy/content/test.html',
             'widgy/core_tests/test.html',
-            'widgy/models/test.html',
+            'widgy/widgy/test.html',
             'widgy/test.html',
         ])
diff --git a/widgy/models/base.py b/widgy/models/base.py
index 39c8d97..656a5a3 100644
--- a/widgy/models/base.py
+++ b/widgy/models/base.py
@@ -424,6 +424,7 @@ class Content(models.Model):
 
     class Meta:
         abstract = True
+        app_label = 'widgy'
 
     def __init__(self, *args, **kwargs):
         super(Content, self).__init__(*args, **kwargs)
@@ -824,6 +825,10 @@ class Content(models.Model):
         return self.get_attributes() == other.get_attributes()
 
 
+# Otherwise, mixins will inherit Content.Meta and get wierd app_labels
+del Content.Meta
+
+
 class UnknownWidget(Content):
     """
     A placeholder Content class used when the correct one can't be found. For
```

**Comment by gavinwahl** [comment]
I fixed the rest of the contrib failures, this should be good to merge.

**Comment by plee** [comment]
:shipit: :v:

**Comment by rockymeza** [comment]
This is not good to merge:

https://code.djangoproject.com/ticket/24470

**Comment by acatton** [comment]
We should workaround the serialization bug (by manually fixing the migrations) and try to merge this.

**Comment by gavinwahl** [comment]
Added the migrations, this should be good to go

File: widgy/contrib/page_builder/models.py [hunk 1]
```diff
@@ -485,7 +485,7 @@ class Figure(StrDisplayNameMixin, Content):
     tooltip = _("A figure is a self-contained piece of content. It can be used"
                 " to add a caption to an image or a video for example.")
 
-    position = models.CharField(default=lambda: ugettext('center'), verbose_name=_('position'), max_length=50, choices=[
+    position = models.CharField(default=_('center'), verbose_name=_('position'), max_length=50, choices=[
```
Comment (acatton) [inline widgy/contrib/page_builder/models.py]: Pure question, not a remark. I don't understand this change. Why was it a lambda? Why isn't it a lambda anymore?

Comment (gavinwahl) [inline widgy/contrib/page_builder/models.py]: A lambda can't be serialized into a migration but a ugettext_lazy object can

**Pull request merged**